src/loader.c

```diff
@@ -987,16 +985,28 @@ int get_cgroup_data(const char *pid_cgroup, char *pod_uid, char *container_id,
     LOGGER(4, "can't find memory cgroup from %s", pid_cgroup);
     goto DONE;
   }
+  LOGGER(4, "get cgroup path %s", cgroup_ptr);
+
 
   /**
    * find container id
+   *
+   * if cgroup is cgroupfs, cgroup pattern should be like
+   *
+   * docker
+   * /kubepods/besteffort/pod27882189_b4d9_11e9_b287_ec0d9ae89a20/docker-4aa615892ab2a014d52178bdf3da1c4a45c8ddfb5171dd6e39dc910f96693e14
+   *
+   * containerd
+   * /system.slice/containerd.service/kubepods-besteffort-pod019c1fe8_0d92_4aa0_b61c_4df58bdde71c.slice:cri-containerd:9e073649debeec6d511391c9ec7627ee67ce3a3fb508b0fa0437a97f8e58ba98
```

Comment:
Which version of kubernetes generates this cgroup pattern?

Reply:
try v1.21.x with containerd and cgroupfs.